### ISSUE TITLE
请问RVM的MNN模型是如何转换的？

### ISSUE THREAD

**Issue by TCBocean** [DefTruth/lite.ai.toolkit#165]
最近我也在做RVM的部署，但是我从原作者的工程中导出ONNX再转MNN，运行出来的结果会有较大差距，但是用你的模型就没问题。所以想请教一下，这个RVM的模型是如何转换出来的呢？

**Comment by DefTruth** [comment]
hi~ 我是自己重新转了静态的onnx，然后过了onnxsim一遍。之后再转的MNN。主要转静态的onnx比较复杂，需要修改比较多RVM原作者源码。建议你可以参考下我另一个的案例项目的issue.
* [转onnx时，有了rxi的值之后应该怎样设置呢？](https://github.com/DefTruth/RobustVideoMatting.lite.ai.toolkit/issues/12)
* [固定downsample，这里是不是也需要改动？](https://github.com/DefTruth/RobustVideoMatting.lite.ai.toolkit/issues/13)

另外如果你需要转基于resnet backbone的模型，还需要修改hardsigmoid，可以参考我在RVM issue里面的回复:  
* [hardsigmoid replacement ](https://github.com/PeterL1n/RobustVideoMatting/issues/47)  
> So your inference framework does not support `HardSigmoid`. But no worries, hard sigmoid can be replaced with more primitives ops.
> 
> You can change them to formula below. The actual export code you need to figure out. Shouldn't be difficult.
> 
> ```
> Hardsigmoid(x) = x.clamp(0, 6).add(3).div(6).
> Hardswish(x) = x * Hardsigmoid(x)
> ```

it seems that should change  `Hardsigmoid` to `x.add(3).clamp(0,6).div(6)`, not `x.clamp(0, 6).add(3).div(6)`:
```python
import torch
from torch.nn import functional as F

if __name__ == "__main__":
    x = torch.FloatTensor([-4, -3. -2., -1., 0., 1., 2., 3., 4.])
    hs1 = x.add(3).clamp(0, 6).div(6)
    hs2 = F.hardsigmoid(x)
    rest = hs1 - hs2
    print(hs1)
    print(hs2)
    print(rest)
```  
log:  
```shell
tensor([0.0000, 0.0000, 0.3333, 0.5000, 0.6667, 0.8333, 1.0000, 1.0000])
tensor([0.0000, 0.0000, 0.3333, 0.5000, 0.6667, 0.8333, 1.0000, 1.0000])
tensor([0., 0., 0., 0., 0., 0., 0., 0.])
```  
the result of RVM is not right if i change  `Hardsigmoid` to `x.clamp(0, 6).add(3).div(6)`, but `x.add(3).clamp(0,6).div(6)` work fined. the official implementation of  F.hardsigmoid in pytorch is:  
![image](https://user-images.githubusercontent.com/31974251/136692083-28d19158-d440-4076-9eae-f669489ea08a.png)
i think we should add(3) first and then clamp(0, 6) with post div(6), it make x -> (x+3):  
> 0, if (x+3)<=-3+3=0
> 1, if (x+3)>=+3+3=6
> (x+3)/6, otherwise

**Comment by TCBocean** [comment]
好的，非常感谢你的回答 @DefTruth
